### ISSUE TITLE
Shorewall no longer supported in EPEL on CentOS

### ISSUE THREAD

**Issue by davidfokkema** [HiSPARC/publicdb#159]
Currently on tietar Shorewall is installed from the EPEL repository:
```
shorewall.noarch                           4.5.4-1.el6                  @epel
shorewall-core.noarch                      4.5.4-1.el6                  @epel
```
However, those packages have since been removed from EPEL because the maintainer resigned and nobody picked it up. We can compile Shorewall ourselves, or switch to another firewall. That is some work and not much fun. Currently it is not possible to build a new VPN server and ``vagrant up vpn`` will fail.

**Comment by kaspervd** [comment]
That's unfortunate! This is an issue. Is it possible to use Ansible to download and install Shorewall without relying on EPEL? If not, installing Shorewall manually is probably easier than setting up a new firewall. Is it a problem if we have one non-working (or at least incomplete) vagrant box?

**Comment by davidfokkema** [comment]
It is not *really* a problem if the vagrant box does not work. However, we use ansible to keep the servers in shape and to install a new one if necessary. If Tietar breaks now, we get to keep the pieces.

To fix this, we need ansible to download the shorewall source to /usr/local/src, to install the build dependencies, and to compile and install it. Not very much work, but it has to be done. I'd like to concentrate on Pique and Frome first, however.

**Comment by tomkooij** [comment]
We can also just build a github repo that uses Travis-CI to build `shorewall` on CentOS (and maybe contribute it to `EPEL`).

But so much other things to do...

**Comment by davidfokkema** [comment]
@tomkooij Indeed, so much things to do. Let's not start being a maintainer for EPEL packages.

**Comment by kaspervd** [comment]
Let's go for the download source, compile and install approach and hope the source URL stays te same. I will look into this.

**Comment by kaspervd** [comment]
Update:
I'm able to download the shorewall .rpm file, install it and copy the new configuration file using ansible but somehow I cannot start shorewall. Also, although rpm succesfully installs shorewall it is not listed if I do `rpm -qa`. I'll try to investigate further.

**Comment by davidfokkema** [comment]
Do you have a public PR?

**Comment by kaspervd** [comment]
See the new ansible [file](https://github.com/HiSPARC/publicdb/blob/fixing_shorewall/provisioning/shorewall-firewall.yml)

**Comment by davidfokkema** [comment]
Ah yes, you installed the source rpm. That means the source code is now installed, not the package. You still have to build it, or find a compiled rpm.

**Comment by davidfokkema** [comment]
Partially fixed by 15c009b166977ae7d473f440b49df6bede9cc3aa. Still doesn't register a service with CentOS. Maybe install packages more tailored for RedHat-like systems?

**Comment by kaspervd** [comment]
Thanks! It should work now (https://github.com/HiSPARC/publicdb/commit/99062c6137fb5a4e8aa28bbf4c4bf8ab0d651010). Ansible warned me:

> [WARNING]: Consider using 'become', 'become_method', and 'become_user' rather
than running sudo

Is it OK to simply use sudo?

**Comment by kaspervd** [comment]
I used 'become' anyway, better safe than sorry.

**Comment by kaspervd** [comment]
Unfortunately, the previously used download source is not available anymore. A newer version was placed there but this one yields installation errors. I now use the old version (5.1.2-3) from another download source but somehow it won't allow shorewall to run as a service. Executing shorewall with ansible using `command: shorewall start` does work. Does anyone advice against running shorewall using `command` instead of `service`?

**Comment by davidfokkema** [comment]
Hm. This is annoying. This means an additional command to check and verify everything is working correctly. Furthermore, a service is started on each boot cycle. What happens when the machine is rebooted in its current state? Will it revert to a state of navel gazing?

**Comment by kaspervd** [comment]
I'm afraid you're correct about the navel gazing.. I'll try to figure out why it's not recognised as a service anymore.

**Comment by davidfokkema** [comment]
Pity. Either that, or figure out how to rebuilt the firewall using supported tools...

**Comment by 153957** [comment]
It is supported for CentOS 7.

**Comment by 153957** [comment]
Will be fixed in #230
